### PR TITLE
[IBM]Support private endpoints

### PR DESCRIPTION
**Pull request by praveingk**

### CODE REVIEW

File: pkg/ibm/resource.go [hunk 1]
```diff
@@ -433,6 +446,195 @@ func (c *ResourceClusterType) GetVPC() (*vpcv1.VPCReference, error) {
...
+		}
+		if *res.LifecycleState == endpointReadyState {
+			return true, nil
+		}
+		time.Sleep(sleepDuration)
+	}
+	return false, fmt.Errorf("endpoint gateway ID %v failed to launch within the alloted time", e.ID)
+}
+
+// CreateResource create an instance
```
Comment (smcclure20) [inline pkg/ibm/resource.go:504]: Typo: instance --> private endpoint

File: pkg/ibm/resource.go [hunk 2]
```diff
@@ -433,6 +446,195 @@ func (c *ResourceClusterType) GetVPC() (*vpcv1.VPCReference, error) {
...
+		return false, err
+	}
+
+	// add cluster's CRN and region to search attributes
+	resourceQuery.CRN = clusterCRN
+	if region != "" {
+		resourceQuery.Region = region
+	}
+
+	// look for cluster with the specified CRN in the specified namespace.
```
Comment (smcclure20) [inline pkg/ibm/resource.go:588]: typo: cluster --> private endpoint

File: pkg/ibm/resource.go [hunk 3]
```diff
@@ -433,6 +446,195 @@ func (c *ResourceClusterType) GetVPC() (*vpcv1.VPCReference, error) {
...
+
+// IsInNamespace checks if the private endpoint is in the namespace
+func (e *ResourcePrivateEndpointType) IsInNamespace(namespace, region string) (bool, error) {
+	resourceQuery := resourceQuery{}
+	clusterCRN, err := e.getCRN()
+	if err != nil {
+		return false, err
+	}
+
+	// add cluster's CRN and region to search attributes
```
Comment (smcclure20) [inline pkg/ibm/resource.go:582]: typo: cluster --> private endpoint

File: pkg/ibm/resource.go [hunk 4]
```diff
@@ -433,6 +446,195 @@ func (c *ResourceClusterType) GetVPC() (*vpcv1.VPCReference, error) {
...
+
+	resp := ResourceResponse{Name: *endpointGateway.Name, URI: e.createURI(*e.client.resourceGroup.ID, *zone, *endpointGateway.ID), IP: reservedIP}
+
+	return &resp, nil
+}
+
+// IsInNamespace checks if the private endpoint is in the namespace
+func (e *ResourcePrivateEndpointType) IsInNamespace(namespace, region string) (bool, error) {
+	resourceQuery := resourceQuery{}
+	clusterCRN, err := e.getCRN()
```
Comment (smcclure20) [inline pkg/ibm/resource.go:577]: cluster or private endpoint? (or something similar)

**Review by smcclure20 — CHANGES_REQUESTED** [review]
Thanks for this! Mostly just some nits. 

Also noting that once this is in, we should include some documentation on what the payload should exactly look like for requests.

File: pkg/ibm/plugin_test.go [hunk 5]
```diff
@@ -366,18 +374,38 @@ func getFakeIBMServerHandler(fakeIBMServerState *fakeIBMServerState) http.Handle
 							{
 								ID:            core.StringPtr(r.URL.Query().Get("vpc.id")),
 								Ipv4CIDRBlock: &cidrBlock,
+								Zone: &vpcv1.ZoneReference{
+									Name: core.StringPtr(fakeZone),
+								},
 							},
 						}
 					}
+				} else {
```
[the remark this answers is not in the document]
Comment (praveingk) [reply pkg/ibm/plugin_test.go:383]: Agree. Will change to have a dependency with creation of subnet, which is common for cluster and vpe.

**Review by smcclure20 — APPROVED** [review]

**Pull request merged**